legend-pure-core/legend-pure-m4/src/test/java/org/finos/legend/pure/m4/coreinstance/primitive/date/TestPureDate.java

```diff
@@ -55,6 +55,25 @@ public void testFormatWithTimeZoneShift()
         Assert.assertEquals("2014-01-01 02:01:01.070+0100", date.format("[CET]yyyy-MM-dd HH:mm:ss.SSSZ"));
         Assert.assertEquals("2014-01-01 02:01:01.070 CET", date.format("[CET]yyyy-MM-dd HH:mm:ss.SSS z"));
         Assert.assertEquals("2014-01-01 02:01:01.070+01", date.format("[CET]yyyy-MM-dd HH:mm:ss.SSSX"));
+        Assert.assertEquals("2014-01-01 02:01:01.070 CET", date.format("[\"CET\"]yyyy-MM-dd HH:mm:ss.SSS z"));
+
+        // Longer time zone names
+        Assert.assertEquals("2014-01-01 01:01:01.070+0000", date.format("[Europe/Lissabon]yyyy-MM-dd HH:mm:ss.SSSZ"));
+        Assert.assertEquals("2014-01-01 01:01:01.070 Europe/Lissabon", date.format("[Europe/Lissabon]yyyy-MM-dd HH:mm:ss.SSS z"));
+        Assert.assertEquals("2014-01-01 01:01:01.070Z", date.format("[Lissabon]yyyy-MM-dd HH:mm:ss.SSSX"));
+        Assert.assertEquals("2014-01-01 01:01:01.070 Europe/Lissabon", date.format("[\"Europe/Lissabon\"]yyyy-MM-dd HH:mm:ss.SSS z"));
+
+        Assert.assertEquals("2014-01-01 00:01:01.070-0100", date.format("[Atlantic/Azores]yyyy-MM-dd HH:mm:ss.SSSZ"));
+        Assert.assertEquals("2014-01-01 00:01:01.070 Atlantic/Azores", date.format("[Atlantic/Azores]yyyy-MM-dd HH:mm:ss.SSS z"));
+        Assert.assertEquals("2014-01-01 00:01:01.070-01", date.format("[Atlantic/Azores]yyyy-MM-dd HH:mm:ss.SSSX"));
+
+        // Time zone is illegal after date or time has been formatted and written out
+        Assert.assertEquals("Time zone can only be set at the beginning of the format string",
+                Assert.assertThrows(IllegalArgumentException.class, () -> date.format("yyyy-MM-dd HH:mm:ss.SSS [CET]z")).getMessage());
+        Assert.assertEquals("Time zone can only be set at the beginning of the format string",
+                Assert.assertThrows(IllegalArgumentException.class, () -> date.format("yyyy-MM-dd [CET]HH:mm:ss.SSSz")).getMessage());
+        Assert.assertEquals("Time zone can only be set at the beginning of the format string",
+                Assert.assertThrows(IllegalArgumentException.class, () -> date.format("yyyy-MM-dd[CET] HH:mm:ss.SSSz")).getMessage());
```

Comment:
Can you add some cases here demonstrating that multiple time zones would be illegal, regardless of whether they come at the start of the string? The invalid format string from the deleted testFormatWithMultipleTimeZones should still throw an exception (though possibly a different one). Also, a format string like "[CET] [CST] yyyy-MM-dd HH:mm:ss.SSSz" should throw an exception.

Reply:
I made this assumption:

    if (i > 1)
                    {
                        throw new IllegalArgumentException("Time zone can only be set at the beginning of the format string");
                    }

If time zone is not set at the very beginning of the string, it simply fails.

Perhaps this is too rigid?
It doesn't accept preceeding whitespace.

Reply:
I thought about that, but I think it's fine to make it that rigid. It certainly makes the logic easier and the requirement clearer.

Nonetheless, I think we should still have test cases asserting that multiple time zones will result in an exception. Even if it seems like it's implied by other asserts, I think it's good to have test cases that explicitly assert it.

Reply:
Returned in commit 98bbe1bc1.